### PR TITLE
Avoid sending empty headers and cookie to FFMPEG

### PR DESCRIPTION
**Pull request by geffrak**
### Linked issues

N/A

### Summarize your change.

1) Avoid submitting cookies and headers parameters to FFMPEG if they are empty strings... it doesn't like it.
`No trailing CRLF found in HTTP header` -> empty header. 

2) Use the proper delimiter of `\r\n` for headers because that's the spec.

### Describe the reason for the change.

There was one warning line per web media loaded. 

### Describe what you have tested and on which operating system.

Tested on MacOS, but not platform-specific. 

### Add a list of changes and note any needing special attention during the review.

The change is in 44c9bdc3ea77ff1a3dba83b7beeff130a8db624b, the other commit is just automatic formatting from my editor. 

### If possible, provide screenshots.

N/A

### CODE REVIEW

File: src/lib/ip/IPBaseNodes/FileSourceIPNode.cpp [hunk 1]
```diff
@@ -2547,52 +2646,63 @@ FileSourceIPNode::openMovieTask(const string& filename,
...
+          }
+
+          TwkMediaLibrary::HTTPHeaderVector headers = api->httpHeaders();
+          if( !headers.empty() )
+          {
+            ostringstream headersStm;
+
+            for( size_t h = 0, size = headers.size(); h < size; ++h )
+            {
+              if( h > 0 ) headersStm << "\r\n";
```
Comment (geffrak) [inline src/lib/ip/IPBaseNodes/FileSourceIPNode.cpp:2698]: copy paste leftover on the ostrinstream name.

File: src/lib/ip/IPBaseNodes/FileSourceIPNode.cpp [hunk 2]
```diff
@@ -2547,52 +2646,63 @@ FileSourceIPNode::openMovieTask(const string& filename,
...
+      for( int n = 0; n < nodes.size(); n++ )
+      {
+        const TwkMediaLibrary::Node* node = nodes[n];
+        if( const TwkMediaLibrary::MediaAPI* api =
+                TwkMediaLibrary::api_cast<TwkMediaLibrary::MediaAPI>( node ) )
         {
-            const TwkMediaLibrary::Node* node = nodes[n];
-            if (const TwkMediaLibrary::MediaAPI* api = TwkMediaLibrary::api_cast<TwkMediaLibrary::MediaAPI>(node))
+          TwkMediaLibrary::HTTPCookieVector cookies = api->httpCookies();
+          if( !cookies.empty() )
```
Comment (geffrak) [inline src/lib/ip/IPBaseNodes/FileSourceIPNode.cpp:2676]: Only sets the cookies param if there are cookies to be set.

File: src/lib/ip/IPBaseNodes/FileSourceIPNode.cpp (hunk 1 above)
Comment (geffrak) [reply src/lib/ip/IPBaseNodes/FileSourceIPNode.cpp:2698]: The spec expects \r\n for headers.

File: src/lib/ip/IPBaseNodes/FileSourceIPNode.cpp [hunk 3]
```diff
@@ -2547,52 +2646,63 @@ FileSourceIPNode::openMovieTask(const string& filename,
...
-                    if (h > 0) cookieStm << "\n";
-                    headersStm << headers[h].name << ": " << headers[h].value;
-                
-                }
-                m_inparams.push_back(StringPair("headers", headersStm.str()));
+            m_inparams.push_back( StringPair( "cookies", cookieStm.str() ) );
+          }
+
+          TwkMediaLibrary::HTTPHeaderVector headers = api->httpHeaders();
+          if( !headers.empty() )
```
Comment (geffrak) [inline src/lib/ip/IPBaseNodes/FileSourceIPNode.cpp:2692]: Only sets the headers param if there are headers to be set.

**Review by rogernelson — APPROVED** [review]

**Review by bernie-laberge — APPROVED** [review]

**Pull request merged**